python/lsst/daf/persistence/persistence.cc

```diff
@@ -7,24 +7,32 @@
 #include "lsst/daf/persistence/Storage.h"
 #include "lsst/daf/persistence/LogicalLocation.h"
 
-using namespace lsst::daf::persistence;
+#include "lsst/daf/persistence/pybind11/readProxy.h"
 
 namespace py = pybind11;
 
-PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
+namespace lsst {
+namespace daf {
+namespace persistence {
 
 PYBIND11_PLUGIN(_persistence) {
     py::module mod("_persistence", "Access to the classes from the daf_persistence persistence library");
 
-    py::class_<Persistence, std::shared_ptr<Persistence>, lsst::daf::base::Citizen> cls(mod, "Persistence");
+    py::class_<python::ReadProxyBase, std::shared_ptr<python::ReadProxyBase>>(mod, "ReadProxyBase")
+        .def(py::init<>())
+        .def_readwrite("subject", &python::ReadProxyBase::subject);
```

Comment:
Could we call this `__subject__` and remove `__subject__` from `ReadProxy.__slots__`?  It's possible we might also have to replace `object.__getattribute__` in `ReadProxy` with `ReadProxyBase.__getattribute__` (I don't know if pybind11 overrides that, but it's conceivable that it does for e.g. static properties).

In any case, I do think it'd be worth some effort to avoid having the proxy introduce another attribute, *especially* a non-dunder one.